src/bigquery.rs

```diff
@@ -154,13 +154,19 @@ impl TranslateFrom<ast::Tag> for Tag {
             ast::Type::Map(map) => {
                 let key = Tag::translate_from(*map.key.clone(), context).unwrap();
                 let value = match Tag::translate_from(*map.value.clone(), context) {
-                    Ok(tag) => tag,
-                    Err(_) => return Err(fmt_reason("untyped map value")),
+                    Ok(tag) => Some(tag),
+                    // Err is only reachable when context.resolve_method is Drop
+                    Err(_) => match context.allow_maps_without_value {
+                        true => None,
+                        false => return Err(fmt_reason("untyped map value")),
+                    },
                 };
-                let fields: HashMap<String, Box<Tag>> = vec![("key", key), ("value", value)]
-                    .into_iter()
-                    .map(|(k, v)| (k.to_string(), Box::new(v)))
-                    .collect();
+                let fields: HashMap<String, Box<Tag>> =
+                    vec![Some(("key", key)), value.map(|v| ("value", v))]
```

Comment:
This could have been a `value.unwrap()`, too.

Reply:
thx, didn't realize I could `.unwrap()` an `Option`